Modules/posixmodule.c

```diff
@@ -5632,8 +5646,37 @@ os_posix_spawnp_impl(PyObject *module, path_t *path, PyObject *argv,
 }
 #endif /* HAVE_POSIX_SPAWNP */
 
+#ifdef HAVE_RTPSPAWN
+static intptr_t
+_rtp_spawn(int mode, const char *rtpFileName, const char *argv[],
+               const char  *envp[])
+{
+     RTP_ID rtpid;
+     int status;
+     pid_t res;
+
+     if (envp) {
+         rtpid = rtpSpawn(rtpFileName, argv, envp,
+                          100, 0x1000000, 0, VX_FP_TASK);
```

Comment:
What are these hardcoded constants? Can you please add a comment to explain them?

Reply:
The prototype for rtpSpawn is as follows. BTW, RTP stands for real time process.

RTP_ID rtpSpawn
    (
    const char * rtpFileName,	/* Null terminated path to executable */
    const char * argv[],	/* Pointer to NULL terminated argv array */
    const char * envp[],	/* Pointer to NULL terminated envp array */
    int		 priority,	/* Priority of initial task */
    size_t	 uStackSize,	/* User space stack size for initial task */
    int		 options,	/* The options passed to the RTP */
    int		 taskOptions	/* Task options for the RTP's initial task */
    )
	
<rtpFileName>: this parameter specifies the executable path.
<argv>: this parameter points to the string array for arguments of executable. The argv[0] is also the executable pathname.
<envp>: this parameter points to the environment array used to pass specific RTP environment variable settings to the application.
<priority>: this parameter sets the priority of the RTP's initial task.
<uStackSize>: this parameter sets the size of the initial task's user-side stack.
<options>: this parameter specify the behavior of the RTP, for example, stoping at the startup for debugging, not inheriting the CPU affinity of the calling task, etc.
<taskOptions>: this parameter allows to pass options to the initial task created with the RTP.

VxWorks is priority-preempted OS. We set the priority to 100 here after we think this is a proper value.

Reply:
Oh, 0x1000000 is the hardcoded stack size? The OS doesn't provide a good default as a constant?

Same question for the priority.

I have no preference, it's just that I prefer to avoid hardcoded values whenever possible ;-)

Reply:
If  uStackSize is left null on VxWorks this size is set to the default value (0x4000 bytes for 32bit CPU variants and 0x8000 for 64bit CPU variants). In our internal discussion, guys think the default value seems not sufficient for Python. So we set it 0x1000000 here. But indeed the hardcoded value is sort of uncomfortable. Creating a header file, for example, pyvxconfig.h like pymacconfig.h does, defines some macros for default values then include it in Python.h? Or just simply define some macros at the top of this file posixmodule.c? Could you give us some advice? Thanks.

Reply:
Oops, forgot explaining the priority. On VxWorks, we have to take care of it byself. Care should be taken in setting a priority appropriate for an application. Leaving this parameter set to zero would create an initial task of he highest priority in VxWorks, possibly disturbing the functioning the
rest of the system.

Reply:
I see, it's more a practical solution. In that, just add a comment. Something like:

```
/* priority=100 and uStackSize=16 MiB (0x1000000).
   uStackSize=0 cannot be used, the default stack size is too small for Python.  */
```

Reply:
Thanks for your idea.